wsg_50_driver/include/wsg_50/optical_weiss_finger.h

```diff
@@ -0,0 +1,341 @@
+    */
+    bool restart_reading();
+
+    /**
+     * @brief clear_sample_buffer: Remove previous samples from the sample buffer
+     * @return Whether or not the sample buffer was successfully cleared
+    */
+    bool clear_sample_buffer();
+
+    /**
+     * @brief get_sample: Get the latest data sample
+     * @return The latest data sample
+    */
+    wsg_50_common::WeissFingerData get_sample();
+
+    /**
+     * @brief get_latest_samples: Get the most recent n samples, ordered from newest to oldest
+     * @param n_msgs: The number of samples to get. Will return all samples if less than or equal to zero
+     * @param buff: Container for the n samples
+     * @return True if at least one sample was returned, otherwise False
+    */
+    bool get_latest_samples(unsigned int n_msgs, std::vector<wsg_50_common::WeissFingerData>& buff);
+
+    /**
+     * @brief hand_width_to_finger_tip_separation: Computes the distance from this finger to the target on
+     *        the other finger.
+     * @param width: The current width of the hand
+     * @return The distance between the surface of this finger and the target on the other fingertip
+    */
+    double hand_width_to_finger_tip_separation(double width);
+private:
+
+    ros::NodeHandle nh_; // Node handle used to create publisher of data
+    bool initialized_; // Status flag, indicates if object has been initialized
+    bool can_calibrate_; // Status flag, indicates if object is able to perform calibration
+    std::deque<wsg_50_common::WeissFingerData> data_buff_; // Contains the most recent data samples
+    std::mutex data_buffer_mutex_; // Mutex that controls access to data_buff_
+    std::thread* read_finger_thread_; // Thread for reading data from the finger
+    bool read_finger_alive_; // Once thread has started, will shutdown if this is false
+    bool paused_; // Indicates if data is being read from the finger
+    std::string pub_topic_; // Topic to publish data to
+
+    unsigned int finger_id_; // The id of this finger. Should be 0 or 1
+    unsigned int devices_; // The number of optical sensors that this finger contains
+    std::vector<uint16_t> reg_addrs_; // Addresses of the optical sensor registers to read
+    std::vector<uint8_t> reg_lengths_; // Lengths of the optical sensor registers to read
+    double finger_read_rate_; // The rate at which data from the finger should be read
+    int data_buff_max_size_; // The maximum size of the data buffer
+    double sensor_to_surface_mm_; // The distance between the top of the optical sensor and the surface of the finger
+    double target_surface_offset_mm_; // The offset of the target (during calibration) from the surface of the other finger
+    std::string calibration_path_; // The file to load/save calibration from/to
+    int calib_see_through_samples_; // The number of samples to collect when determining the minimum detection threshold for the optical sensors
+    int calib_see_through_target_; // Not actually used
```

Comment:
Out of curiosity, why is this here? (I'm just looking at the comment right now).

Reply:
All of the calib_*_target variables specify how far away the fingertips should be from each other when doing that particular calibration. I think I put calib_see_through_target_ in for symmetry, but it's not actually necessary for this particular calibration.